tests/lib/test_openstack_flavor.py

```diff
@@ -0,0 +1,140 @@
+        self.instance.get_missing_flavors(source_cloud, dest_cloud)
+        self.instance.list_flavor.assert_has_calls(
+            [call(source_cloud), call(dest_cloud)],
+            any_order=False,  # Order matters here
+        )
+
+    def test_get_missing_flavors_get_diff(self):
+        """
+        Test getting the difference between list of flavor names
+        """
+        mock_source_names = [NonCallableMock() for _ in range(3)]
+        for i, obj in enumerate(mock_source_names):
+            obj.name = i
+
+        mock_dest_names = [NonCallableMock() for _ in range(3)]
+        for i, obj in enumerate(mock_dest_names):
+            obj.name = i + 2
+
+        mock_side_effect = [mock_source_names, mock_dest_names]
+
+        self.instance.list_flavor = Mock(side_effect=mock_side_effect)
+        return_diff = self.instance.get_missing_flavors(
+            NonCallableMock(), NonCallableMock()
+        )
+        assert [0, 1] == return_diff
+
+    def test_create_flavor(self):
+        """
+        Test we can create a new flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor_data = NonCallableMock()
+
+        self.instance.create_flavor(mocked_cloud_name, mocked_flavor_data)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
+
+        self.api.create_flavor.assert_called_once_with(
+            name=mocked_flavor_data.name,
+            ram=mocked_flavor_data.ram,
+            vcpus=mocked_flavor_data.vcpus,
+            disk=mocked_flavor_data.disk,
+            flavorid="auto",
+            ephemeral=mocked_flavor_data.ephemeral,
+            swap=mocked_flavor_data.swap,
+            rxtx_factor=mocked_flavor_data.rxtx_factor,
+            is_public=mocked_flavor_data.is_public,
+        )
+
+    def test_empty_dict(self):
+        """
+        Test that method set_flavor_specs not called when empty_dictionary passed
+        """
+        empty_flavor_spec = {}
+        mocked_cloud_name = NonCallableMock()
+        flavor_id = NonCallableMock()
+        self.instance.set_flavor_specs(mocked_cloud_name, flavor_id, empty_flavor_spec)
+        self.api.set_flavor_specs.assert_not_called()
+
+    def test_get_specs(self):
+        """
+        Test getting the extra specs from a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor = NonCallableMock()
+
+        self.instance.get_flavor(mocked_cloud_name, mocked_flavor)
+        self.instance.get_flavor_specs(mocked_cloud_name, mocked_flavor)
+
+    def test_set_specs(self):
+        """
+        Test setting the extra specs for a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        flavor_id = NonCallableMock()
+        extra_specs = NonCallableMock()
+
+        self.instance.set_flavor_specs(mocked_cloud_name, flavor_id, extra_specs)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
```

Comment:
Ditto, we're checking the connection opens but not the set_specs method is called